packages/backend/src/api/jobScraping/scraping.ts

```diff
@@ -0,0 +1,349 @@
+  return jobs;
+};
+
+export const remocateScrape = async () => {
+  const browser = await puppeteer.launch({ headless: false, defaultViewport: { width: 1700, height: 3000 } });
+  const page = await browser.newPage();
+
+  page.goto(
+    'https://www.remocate.app/?Search=developer&Location=%F0%9F%93%8D+Any+Location&Category=%F0%9F%8C%9F+Any+Category#job-board'
+  );
+
+  await page.waitForSelector('.board-list.w-dyn-items');
+  const jobLinks = await page.$$('.w-dyn-item[role="listitem"] a.job-card.w-inline-block');
+
+  const links = [];
+  for (const linkElement of jobLinks) {
+    const href = await linkElement.evaluate((el) => el.href);
+    links.push(href);
+  }
+
+  const jobs = [];
+
+  for (const link of links) {
+    await page.goto(link);
+    const html = await page.content();
+    const $ = cheerio.load(html);
+
+    const title = $('.top-title-job').text().trim();
+    const description = $('.text-rich-text > ul').text();
+    const { languages, frameworks } = filterLanguagesAndFrameworks(description);
+    const applicationLink = await shortenLink(link);
+
+    jobs.push({ title, languages, frameworks, applicationLink });
+  }
+  await browser.close();
+  return jobs;
+};
+
+export const expertiseRecruitmentScrape = async () => {
+  const browser = await puppeteer.launch({ headless: false, defaultViewport: { width: 1700, height: 3000 } });
+  const page = await browser.newPage();
+
+  page.goto('https://jobs.expertiserecruitment.com/jobs/Careers');
+
+  await page.waitForSelector('lyte-checkbox[lt-prop-value="Technology"]');
+
+  await page.evaluate(() => {
+    const checkbox = document.querySelector('lyte-checkbox[lt-prop-value="Technology"]') as HTMLElement;
+    if (checkbox && !checkbox.classList.contains('lyteChecked')) {
+      checkbox.click();
+    }
+  });
+  await page.waitForSelector('.posR');
+
+  const jobLinks: any[] = [];
+
+  const jobListings = await page.$$('.cw-filter-joblist');
+  for (const link of jobListings) {
+    const href = await link.$eval('h3:nth-child(1) > a:nth-child(1)', (el) => el.href);
+    jobLinks.push(href);
+  }
+
+  const jobs: any[] = [];
+
+  for (const link of jobLinks) {
+    await page.goto(link);
+    const html = await page.content();
+    const $ = cheerio.load(html);
+
+    const title = $('.cw-jobheader-info > h1:nth-child(2)').text().trim();
+    const description = $('#spandesc > ul:nth-child(2) > li , #spanreq > ul:nth-child(2) > li').text().trim();
+
+    const { languages, frameworks } = filterLanguagesAndFrameworks(description);
+
+    const applicationLink = await shortenLink(link);
+    jobs.push({ title, languages, frameworks, applicationLink });
+  }
+  return jobs;
+};
+
+export const smartRecruitersScrape = async () => {
+  const browser = await puppeteer.launch({ headless: false, defaultViewport: { width: 1700, height: 3000 } });
+  const page = await browser.newPage();
+
+  page.goto('https://jobs.smartrecruiters.com/?keyword=Developer&locationType=remote');
```

Comment:
would be nice to have all the links in constants at the beginning of the file, as they are prone to beign updated based on future updates of the site's url

Reply:
Actually its my first time I scrape or I know that this thing exist so I dont know the best practices yet but Im trying to get the v0 and of course I need your comments and feedback to improve.

Reply:
Ure totally right I'll do it